### PR TITLE
[SYCL] Move free function queries to experimental namespace

### PR DESCRIPTION
**Pull request by alexbatashev**
Test changes are here: https://github.com/intel/llvm-test-suite/pull/350

### CODE REVIEW

File: sycl/include/CL/sycl/ONEAPI/sub_group.hpp [hunk 1]
```diff
@@ -745,5 +754,16 @@ inline sub_group this_sub_group() {
 }
 
 } // namespace ONEAPI
+
+namespace experimental {
+inline ONEAPI::sub_group this_sub_group() {
```
Comment (vladimirlaz) [inline sycl/include/CL/sycl/ONEAPI/sub_group.hpp:759]: can we use sycl::sub_group here as soon as it has been recently changed: https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/sub_group.hpp?
Less deprecation warning is better...

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: sycl/include/CL/sycl/ONEAPI/sub_group.hpp [hunk 2]
```diff
@@ -97,6 +97,13 @@ void store(multi_ptr<T, Space> dst, const vec<T, N> &x) {
 
 } // namespace detail
 
+namespace ONEAPI {
+struct sub_group;
+}
+namespace experimental {
+ONEAPI::sub_group this_sub_group();
```
Comment (vladimirlaz) [inline sycl/include/CL/sycl/ONEAPI/sub_group.hpp:104]:
```suggestion
sycl::sub_group this_sub_group();
```

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: sycl/include/CL/sycl/ONEAPI/sub_group.hpp (hunk 2 above)
Comment (alexbatashev) [reply sycl/include/CL/sycl/ONEAPI/sub_group.hpp:104]: This wouldn't work. `sycl::sub_group` is an alias to `ONEAPI::sub_group`, defined in `sycl/sub_group.hpp`, which in turn includes `sycl/ONEAPI/sub_group.hpp`. So, I'd either reference an undefined symbol or create a cyclic dependency.
I can add one more alias to `sycl/ONEAPI/sub_group.hpp`, if you'd like. It seems it doesn't lead to ODR violation or something.

**Comment by gmlueck** [comment]
Since these functions are a OneAPI extension, I think the namespace should be `::sycl::ext::oneapi::experimental`.  This is consistent with other OneAPI experimental extensions.

**Comment by dm-vodopyanov** [comment]
Makes sense to merge this PR after https://github.com/intel/llvm/pull/4014 which introduces sycl::ONEAPI -> sycl::ext::oneapi transition.

**Comment by alexbatashev** [comment]
I've updated patch to conform to https://github.com/intel/llvm/pull/4014

**Comment by alexbatashev** [comment]
@intel/dpcpp-specification-reviewers @intel/llvm-reviewers-runtime  ping

**Review by s-kanaev — APPROVED** [review]
LGTM

**Pull request merged**